FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,60 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url, completion: { [weak self] result in
+			guard self != nil else { return }
+
+			var completionResult: FeedLoader.Result = .failure(Error.invalidData)
+			defer {
+				completion(completionResult)
+			}
+
+			do {
+				let (data, urlResponse) = try result.get()
+
+				guard urlResponse.statusCode == 200 else { throw Error.invalidData }
+
+				let feedResponse = try JSONDecoder().decode(FeedItemsResponse.self, from: data)
+				let items = feedResponse.items.map(FeedImage.init)
+
+				completionResult = .success(items)
+
+			} catch let error as Error {
+				completionResult = .failure(error)
+			} catch is DecodingError {
+				completionResult = .failure(Error.invalidData)
+			} catch {
+				completionResult = .failure(Error.connectivity)
+			}
+		})
+	}
+}
+
+private struct FeedItemsResponse {
+	let items: [FeedImageDTO]
+}
+
+extension FeedItemsResponse: Codable {
+	struct FeedImageDTO: Codable {
```

Comment:
Same as my previous comment:
```suggestion
	struct FeedImageDTO: Decodable {
```

Reply:
Totally makes sense, changed it in 15c3409